static/js/reducers/dashboard.js

```diff
@@ -13,39 +14,48 @@ import {
   FETCH_PROCESSING,
   FETCH_SUCCESS,
 } from '../actions';
-import type { DashboardState } from '../flow/dashboardTypes';
+import type {
+  DashboardsState,
+  DashboardState,
+} from '../flow/dashboardTypes';
 import type { Action } from '../flow/reduxTypes';
 
-const INITIAL_DASHBOARD_STATE: DashboardState = {
+export const INITIAL_DASHBOARD_STATE: DashboardState = {
   programs: []
 };
 
-export const dashboard = (state: DashboardState = INITIAL_DASHBOARD_STATE, action: Action) => {
+const INITIAL_DASHBOARDS_STATE: DashboardsState = {};
+
+const updateDashboardState = (state, username, update) => (
+  _.merge({}, state, {[username]: update })
```

Comment:
Hmm... maybe something like this would work:

    {...obj, {username: { ...obj.username, ...newFields } } }

Does `...` work for undefined values?